File: Python/time_comparison.py
import numpy as np
import time

from single_threaded import single_threaded
from multi_threaded import multi_threaded


def time_comparison(img: np.ndarray) -> None:
    start_time = time.time()
    single_threaded(img)
    end_time = time.time()
    print(f"Sequential: {end_time - start_time:.5f}s")

    start_time = time.time()
    multi_threaded(img)
    end_time = time.time()
    print(f"Multithreaded: {end_time - start_time:.5f}s")

File: C++/src/time_comparison.cpp
#include "time_comparison.hpp"
#include "single_threaded.hpp"
#include "multi_threaded.hpp"
#include <chrono>
#include <iostream>

namespace time_comparison
{

  void compareExecutionTimes(cv::Mat &img)
  {
    using namespace std::chrono;

    auto startTime = high_resolution_clock::now();
    single_threaded::processImage(img);
    auto endTime = high_resolution_clock::now();
    auto duration = duration_cast<milliseconds>(endTime - startTime).count();
    std::cout << "Sequential time: " << duration << "ms\n";

    startTime = high_resolution_clock::now();
    multi_threaded::processImage(img);
    endTime = high_resolution_clock::now();
    duration = duration_cast<milliseconds>(endTime - startTime).count();
    std::cout << "Concurrent time: " << duration << "ms\n";
  }

} // namespace time_comparison
